src/drivers/interrupt/cortexm_irq_handle.S

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file
+ * @brief
+ *
+ * @author  Alex Kalmuk
+ * @date    27.03.2017
+ */
+
+.text
+.thumb
+.syntax unified
+.global interrupt_handle_enter
+interrupt_handle_enter:
+
+    mov    r0, sp
+    sub    sp, #58
+    str    r0, [sp, #56]
+    stmia  sp, {r0 - r12}
+    str    lr, [sp, #52]
+
+    mov    r0, sp
+    bl     interrupt_handle
+
+.text
+.thumb
+.syntax unified
+.global __irq_trampoline
+__irq_trampoline:
+
+    cpsid  i
+    # r0 is a pointer to struct irq_saved_state
+    ldr    sp,  [r0, #32]
+    ldr    r14, [r0, #36]
+    bx     r14
```

Comment:
Sure, I will comment these two things.

> R14 holds __pending_handle address here? 

No, R14 holds 0xFFFFFFXX value here. Look, it was restored one line above `ldr    r14, [r0, #36]` . That is, `[r0, #36]` points to the initial value, which was loaded in R14 by CPU on irq handler entering. The actual address of __pending_handle was already stored on the stack in cortexm_nvic.c:
`ctx->lr = (uint32_t) __pending_handle;`